### PR TITLE
Fix tls statemachine during tls v1.2 renegotiation

### PR DESCRIPTION
**Pull request by fAuernigg**

### CODE REVIEW

File: src/tls/openssl/tls_tcp.c [hunk 1]
```diff
@@ -229,7 +229,13 @@ static bool recv_handler(int *err, struct mbuf *mb, bool *estab, void *arg)
...
 	if (SSL_state(tc->ssl) != SSL_ST_OK) {
 
-		if (tc->up) {
+		if (tc->up
+#if !defined(LIBRESSL_VERSION_NUMBER)
+			&& SSL_state(tc->ssl) != TLS_ST_CW_CLNT_HELLO
+			&& SSL_state(tc->ssl) != TLS_ST_CW_FINISHED
+			&& SSL_state(tc->ssl) != TLS_ST_SW_SRVR_DONE
+#endif
+		) {
```
Comment (alfredh) [inline src/tls/openssl/tls_tcp.c:238]: it is a bit difficult to read ...

perhaps it is better to declare a bool before the if and use that bool inside the if ?

**Comment by alfredh** [comment]
thanks, this looks much better :)

**Comment by cspiel1** [comment]
Looks good to me.

**Comment by sreimers** [comment]
Are there any security risks by handling/accepting renegotiations?

- https://github.com/google/boringssl/blob/master/PORTING.md#tls-renegotiation
- https://github.com/curl/curl/issues/3258 

Or is this unrelated?

**Comment by cspiel1** [comment]
Maybe we could make it configurable for specific HTTP URL paths to allow TLS 1.2 renegotiation in a hopefully simple and readable way.

**Pull request closed without merging**